File: app/src/main/java/com/stedi/randomimagegenerator/app/view/activity/LifeCycleActivity.java
package com.stedi.randomimagegenerator.app.view.activity;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import com.stedi.randomimagegenerator.app.di.Components;
import com.stedi.randomimagegenerator.app.other.logger.Logger;

import javax.inject.Inject;

public abstract class LifeCycleActivity extends AppCompatActivity {
    private static final boolean LOG = false;

    @Inject Logger logger;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        if (LOG) {
            Components.getAppComponent(this).inject(this);
            logger.log(this, "onCreate");
        }
        super.onCreate(savedInstanceState);
    }

    @Override
    protected void onRestart() {
        if (LOG)
            logger.log(this, "onRestart");
        super.onRestart();
    }

    @Override
    protected void onStart() {
        if (LOG)
            logger.log(this, "onStart");
        super.onStart();
    }

    @Override
    protected void onResume() {
        if (LOG)
            logger.log(this, "onResume");
        super.onResume();
    }

    @Override
    protected void onPause() {
        if (LOG)
            logger.log(this, "onPause");
        super.onPause();
    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        if (LOG)
            logger.log(this, "onSaveInstanceState");
        super.onSaveInstanceState(outState);
    }

    @Override
    protected void onStop() {
        if (LOG)
            logger.log(this, "onStop");
        super.onStop();
    }

    @Override
    protected void onDestroy() {
        if (LOG)
            logger.log(this, "onDestroy");
        super.onDestroy();
    }
}

File: app/src/main/java/com/stedi/randomimagegenerator/app/view/activity/base/LifeCycleActivity.kt
package com.stedi.randomimagegenerator.app.view.activity.base

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import timber.log.Timber

abstract class LifeCycleActivity : AppCompatActivity() {
    private val LOG = false

    override fun onCreate(savedInstanceState: Bundle?) {
        if (LOG) Timber.i("onCreate")
        super.onCreate(savedInstanceState)
    }

    override fun onRestart() {
        if (LOG) Timber.i("onRestart")
        super.onRestart()
    }

    override fun onStart() {
        if (LOG) Timber.i("onStart")
        super.onStart()
    }

    override fun onResume() {
        if (LOG) Timber.i("onResume")
        super.onResume()
    }

    override fun onPause() {
        if (LOG) Timber.i("onPause")
        super.onPause()
    }

    override fun onSaveInstanceState(outState: Bundle) {
        if (LOG) Timber.i("onSaveInstanceState")
        super.onSaveInstanceState(outState)
    }

    override fun onStop() {
        if (LOG) Timber.i("onStop")
        super.onStop()
    }

    override fun onDestroy() {
        if (LOG) Timber.i("onDestroy")
        super.onDestroy()
    }
}